src/test/java/org/terasology/pathfinding/PathfinderSystemTest.java

```diff
@@ -53,9 +45,9 @@ public void updateChunkBeforePathRequests() throws InterruptedException {
         OnChunkLoaded chunkLoadedDummyEvent = new OnChunkLoaded(new Vector3i());
 
         navGraphSystem.chunkReady(chunkLoadedDummyEvent, entityRef);
-        ListenableFuture f1 = pathfinderSystem.requestPath(entityRef, new Vector3i(), Lists.newArrayList(new Vector3i()));
-        ListenableFuture f2 = pathfinderSystem.requestPath(entityRef, new Vector3i(), Lists.newArrayList(new Vector3i()));
-        ListenableFuture f3 = pathfinderSystem.requestPath(entityRef, new Vector3i(), Lists.newArrayList(new Vector3i()));
+        ListenableFuture<?> f1 = pathfinderSystem.requestPath(entityRef, new Vector3i(), Lists.newArrayList(new Vector3i()));
```

Comment:
for my information: what does `<?>` do?

Reply:
The answer can be found here: _[What is the difference between the unbounded wildcard parameterized type and the raw type?](http://www.angelikalanger.com/GenericsFAQ/FAQSections/ParameterizedTypes.html#FAQ303)_

You may need to backtrack through some of the other linked entries there to get a hang of the jargon. Here, `ListenableFuture` was used as a _raw_ generic type, `ListenableFuture<List<Vector3i>>` would be a _parameterized_ type, and `ListenableFuture<?>` is a _wildcard_ parameterized type.

This changed motivated by the IntelliJ inspection "Raw use of parameterized class." The FAQ notes this enables the compiler to catch some cases that would have been warnings otherwise and reject them as errors.